### PR TITLE
cpu/cc2538: vendor header cleanup

### PR DESCRIPTION
**Pull request by haukepetersen**
EDIT
~~cleaned up the vendor header files to make them excludeable to doxygen...~~

cleaned up CC2538 header files and tuned doxygen so there are no more warnings

### CODE REVIEW

File: doc/doxygen/riot.doxyfile [hunk 1]
```diff
@@ -821,6 +821,7 @@ EXCLUDE_PATTERNS       = */board/*/tools/* \
                          */cpu/*/include/instance/* \
                          */cpu/*/include/pio/* \
                          */cpu/*/include/atmel/* \
+                         */cpu/cc2538/include/cc2538*.h \
```
Comment (OlegHahm) [inline doc/doxygen/riot.doxyfile]: I thought @kaspar030 was against this.

**Comment by kaspar030** [comment]
@haukepetersen Are they vendor supplied? They look like written for RIOT, or at least most of them have RIOT-compliant doxygen.

**Comment by haukepetersen** [comment]
They should be vendor supplied, but since TI is not providing any CMSIS style headers, we wrote them ourselves. I know I also argued for not excluding the registers files of the device drivers, but in this case I vote for exclude these headers to be consistent over all CPUs

**Comment by PeterKietzmann** [comment]
I have no real opinion on this. Kaspar seems to have one so I assign him

**Comment by haukepetersen** [comment]
fixed one include issue and squashed.

**Comment by kaspar030** [comment]
> They should be vendor supplied, but since TI is not providing any CMSIS style headers, we wrote them ourselves.

I still think, if we write headers, we should document them (not exclude).

> I know I also argued for not excluding the registers files of the device drivers, but in this case I vote for exclude these headers to be consistent over all CPUs

Whats the aim of this "consistency"? "consistent no doxygen in cpu/include"?

Why don't we consistently move all vendor supplied files into "cpu/x/vendor", so they're easily excluded and the pathname suggests they're not written by us?

**Comment by OlegHahm** [comment]
I still think there's no need for register descriptions in RIOT's API - independent who wrote them. But if we want consistency, we should either have documentation for all register definitions or for none of them.

**Comment by kaspar030** [comment]
@OlegHahm It's not about the register definitions - it's about the file they're in. Register definitions dan just be ```@name "blahdevice register definitions @{ ... @}```.

> if we want consistency, we should either have documentation for all register definitions or for none of them.

We don't want consistency, we want the advantages that consistency gives us.

**Comment by haukepetersen** [comment]
I take this PR from the release, as it seems to me it makes more sense to finish this discussions once and for all before merging/or not. But for this we need little more time -> so after the current release...

**Comment by haukepetersen** [comment]
consensus: all files written by us should not be excluded from doxygen.

Will adapt this PR accordingly

**Comment by haukepetersen** [comment]
done. No doxygen exluding anymore, but fixed all headers...

**Comment by kaspar030** [comment]
ACK when travis agrees.

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
Travis is happy -> go

**Pull request merged**